app/src/Extensions/LastLoginExtension.php

```diff
@@ -0,0 +1,31 @@
+<?php
+namespace SilverStripe\Bambusa\Extensions;
+
+use SilverStripe\Forms\FieldList;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\FieldType\DBDatetime;
+
+/**
+ * Track the last authentication of each user.
+ */
+class LastLoginExtension extends DataExtension
+{
+
+    private static $db = [
+        'LastLogin' => 'Datetime'
+    ];
+
+    public function updateCMSFields(FieldList $fields)
+    {
+        $fields->removeByName('LastLogin');
+    }
+
+    /**
+     * Just before a member gets logged in (and its record gets written), set the LastLogin timestamp to now.
```

Comment:
The member gets written at least two layers of indirection away from this action (not after the logic which calls `$this->extend('beforeMemberLoggedIn')`), so it's a bit of a shakey implementation to save one database write per session (not per page view). Anyway, I'm assuming the write *does* happen although I can't immediately see where in framework.

Reply:
`afterMemberLoggedIn` calls a write.

I also thought that it might be weird to write it right away as if something preventing the login from completing, you would have written the last login date any way.